### PR TITLE
VTX-4445: shrink scheduler state size

### PR DESCRIPTION
**Pull request by fsdvh**
~~Introducing a new execution graph type: `Completed` which will store all necessary info (excluding actual stages as it can be very big), precalculation stage statistics on completion~~

In case of completion moving stage metrics to job status, removing execution graph from the completed job state. Also in some places I was able to get rid of cloning

### CODE REVIEW

**Review by thinkharderdev — APPROVED** [review]
some nits but lgtm

File: ballista/scheduler/src/state/execution_graph.rs [hunk 1]
```diff
@@ -912,33 +975,32 @@ impl ExecutionGraph {
...
+                        partition_id,
+                        executor_id,
+                    },
+                ))
+            }
+        }
+
+        if !tasks.is_empty() {
+            return Some(tasks);
+        }
```
[the remark this answers is not in the document]
Comment (fsdvh) [reply ballista/scheduler/src/state/execution_graph.rs:998]: 😮

File: ballista/scheduler/src/state/execution_graph.rs [hunk 2]
```diff
@@ -1500,15 +1550,15 @@ impl ExecutionGraph {
         }
 
         info!(
```
[the remark this answers is not in the document]
Comment (fsdvh) [reply ballista/scheduler/src/state/execution_graph.rs:1553]: Done

File: ballista/scheduler/src/state/execution_graph.rs [hunk 3]
```diff
@@ -202,97 +257,224 @@ impl ExecutionGraph {
...
+            let mut total_task_duration_ms = 0;
+            for stage in stages.values() {
+                if let ExecutionStage::Successful(stage) = stage {
+                    completed_stages += 1;
+                    for task in stage.task_infos.iter().filter(|t| t.is_finished()) {
+                        total_task_duration_ms += task.execution_time() as u64
+                    }
+                }
+
+                if let ExecutionStage::Running(stage) = stage {
```
[the remark this answers is not in the document]
Comment (fsdvh) [reply ballista/scheduler/src/state/execution_graph.rs]: I don't think we should do that for metrics, maybe it's a good idea to place a warning here

**Pull request merged**